drivers/backup/portworx/portworx.go

```diff
@@ -1040,6 +1061,31 @@ func (p *portworx) UpdateOwnershipRule(ctx context.Context, req *api.RuleOwnersh
 	return p.ruleManager.UpdateOwnership(ctx, req)
 }
 
+func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (string, error) {
+	var backupUID string
+	var totalBackups int
+	bkpEnumerateReq := &api.BackupEnumerateRequest{OrgId: orgID}
+	bkpEnumerateReq.EnumerateOptions = &api.EnumerateOptions{MaxObjects: uint64(enumerateBatchSize), ObjectIndex: 0}
+	for {
+		enumerateRsp, err := p.EnumerateBackup(ctx, bkpEnumerateReq)
+		if err != nil {
+			return backupUID, fmt.Errorf("Failed to enumerate backups for org %s ctx: [%v]", orgID, err)
+		}
+		for _, backup := range enumerateRsp.GetBackups() {
+			if backup.GetName() == backupName {
+				return backup.GetUid(), nil
+			}
+			totalBackups++
+		}
+		if uint64(totalBackups) >= enumerateRsp.GetTotalCount() {
```

Comment:
Just curious, 
Can this check in _line no 1080_ be ever greater than the TotalCount? 
I can understand totalBackups being  less than or it can be equal to GetTotalCount()..

Reply:
The idea is to get all backups as long as uint64(totalBackups) < enumerateRsp.GetTotalCount().
Now in ideal case the breaking condition could have been uint64(totalBackups) == enumerateRsp.GetTotalCount().
But there can be a case where enumerate is happening and meanwhile some delete happened , in that case the 
totalBackups count (as this is being handled by client) can be bigger than enumerateRsp.GetTotalCount() (which is always the fresh counts).